src/com/linkedin/parseq/internal/Continuations.java

```diff
@@ -16,8 +16,9 @@
  * such that {@code actionX} can throw exception or recursively call {@code method}
  * multiple times, without worry about stack overflow.
  * <p>
- * The guarantee is that actions are called in the same order than recursive approch
- * would have called them.
+ * The guarantee is that actions are called in the same order than recursive approach
+ * would have called them. To put it in another way, this class guarantees to traversal the
+ * execution tree in post-order.
```

Comment:
That's the confusing part... I thought it was testing post-order traversal implementation.